In [1]:
import pandas as pd
df = pd.read_json('Electronics.json', lines='TRUE')

 

In [2]:
pd.__version__

u'0.23.4'

In [3]:
import numpy as np

In [4]:
from IPython.display import display

display(df)

asin   helpful  overall  \
0        0528881469    [0, 0]        5   
1        0528881469  [12, 15]        1   
2        0528881469  [43, 45]        3   
3        0528881469   [9, 10]        2   
4        0528881469    [0, 0]        1   
5        0594451647    [3, 3]        5   
6        0594451647    [0, 0]        2   
7        0594451647    [0, 0]        5   
8        0594451647    [0, 0]        4   
9        0594451647    [3, 3]        5   
10       0594481813    [2, 2]        4   
11       0594481813    [0, 0]        5   
12       0594481813    [1, 1]        5   
13       0594481813    [0, 1]        3   
14       0594481813    [2, 2]        5   
15       0594481813    [0, 0]        3   
16       0594481813    [3, 5]        3   
17       0594481813    [2, 2]        4   
18       0972683275    [0, 0]        5   
19       0972683275    [1, 1]        5   
20       0972683275    [0, 1]        5   
21       0972683275    [0, 0]        4   
22       0972683275    [3, 4]        4   
23       0972683275    [0, 0]        5   
24       0972683275    [0, 0]        4   
25       0972683275    [0, 0]        4   
26       0972683275    [7, 9]        5   
27       0972683275    [1, 1]        5   
28       0972683275    [0, 0]        5   
29       0972683275    [0, 0]        5   
...             ...       ...      ...   
1689158  B00L26YDA4    [1, 1]        5   
1689159  B00L26YDA4    [1, 1]        3   
1689160  B00L26YDA4  [10, 15]        5   
1689161  B00L26YDA4    [0, 0]        4   
1689162  B00L3YHF6O    [5, 8]        5   
1689163  B00L3YHF6O    [2, 4]        5   
1689164  B00L3YHF6O    [2, 2]        5   
1689165  B00L3YHF6O    [3, 4]        5   
1689166  B00L3YHF6O   [7, 11]        5   
1689167  B00L3YHF6O    [2, 7]        5   
1689168  B00L3YHF6O   [5, 12]        5   
1689169  B00L3YHF6O    [1, 2]        5   
1689170  B00L3YHF6O    [0, 3]        5   
1689171  B00L3YHF6O    [0, 0]        5   
1689172  B00L3YHF6O    [2, 2]        5   
1689173  B00L3YHF6O    [1, 2]        5   
1689174  B00L3YHF6O    [0, 1]        5   
1689175  B00L3YHF6O    [3, 3]        5   
1689176  B00L3YHF6O    [5, 8]        5   
1689177  B00L3YHF6O    [0, 0]        5   
1689178  B00L3YHF6O    [0, 0]        5   
1689179  B00L3YHF6O    [0, 0]        5   
1689180  B00L3YHF6O  [18, 23]        5   
1689181  B00L3YHF6O    [0, 0]        5   
1689182  B00L3YHF6O    [7, 9]        5   
1689183  B00LGQ6HL8    [1, 1]        5   
1689184  B00LGQ6HL8    [0, 0]        5   
1689185  B00LGQ6HL8    [0, 0]        5   
1689186  B00LGQ6HL8    [0, 0]        5   
1689187  B00LGQ6HL8    [0, 0]        5   

                                                reviewText   reviewTime  \
0        We got this GPS for my husband who is an (OTR)...   06 2, 2013   
1        I'm a professional OTR truck driver, and I bou...  11 25, 2010   
2        Well, what can I say.  I've had this unit in m...   09 9, 2010   
3        Not going to write a long review, even thought...  11 24, 2010   
4        I've had mine for a year and here's what we go...  09 29, 2011   
5        I am using this with a Nook HD+. It works as d...   01 3, 2014   
6        The cable is very wobbly and sometimes disconn...  04 27, 2014   
7        This adaptor is real easy to setup and use rig...   05 4, 2014   
8        This adapter easily connects my Nook HD 7&#34;...  07 11, 2014   
9        This product really works great but I found th...  01 20, 2014   
10       This item is just as was described in the orig...  04 16, 2014   
11       bought for a spare for my 9&#34; Nook HD and i...   05 5, 2014   
12       My son crewed my HD charger cord so I needed a...  06 24, 2013   
13       This is a good beefy 2 amp charger, but it cov...  05 25, 2013   
14       I lost my B&N original cable.  I looked around...   03 9, 2014   
15       It does 2A and charges a DEAD Nook in a few ho...  08 31, 2013   
16       Go to Target or Barnes and Noble instead, and ...  09 18, 2013   
17       Works well, a little pricey I think for a char...  

In [5]:
rows,columns = df.shape

In [6]:
columns

9

In [7]:
rows


1689188

There are 1689188 rows and 9 columns in our data.

In [9]:
# listing the column names

df.columns

Index([u'asin', u'helpful', u'overall', u'reviewText', u'reviewTime',
       u'reviewerID', u'reviewerName', u'summary', u'unixReviewTime'],
      dtype='object')

In [10]:
#Renaming asin to productID

df.rename(columns = {'asin':'productID'}, inplace=True)

In [11]:
df.head()

productID   helpful  overall  \
0  0528881469    [0, 0]        5   
1  0528881469  [12, 15]        1   
2  0528881469  [43, 45]        3   
3  0528881469   [9, 10]        2   
4  0528881469    [0, 0]        1   

                                          reviewText   reviewTime  \
0  We got this GPS for my husband who is an (OTR)...   06 2, 2013   
1  I'm a professional OTR truck driver, and I bou...  11 25, 2010   
2  Well, what can I say.  I've had this unit in m...   09 9, 2010   
3  Not going to write a long review, even thought...  11 24, 2010   
4  I've had mine for a year and here's what we go...  09 29, 2011   

       reviewerID              reviewerName  \
0   AO94DHGC771SJ                   amazdnu   
1   AMO214LNFCEI4           Amazon Customer   
2  A3N7T0DY83Y4IG             C. A. Freeman   
3  A1H8PY3QHMQQA0  Dave M. Shaw "mack dave"   
4  A24EV6RXELQZ63               Wayne Smith   

                                  summary  unixReviewTime  
0                         Gotta have GPS!      1370131200  
1                       Very Disappointed      1290643200  
2                          1st impression      1283990400  
3                 Great grafics, POOR GPS      1290556800  
4  Major issues, only excuses for support      1317254400

Considering 100000 rows of data and the columns productID, overall, reviewText and reviewerID for building our recommender system

In [14]:
df_tmp = df[1:100000]
df_work = df_tmp[['productID','overall','reviewText','reviewerID']].copy()

Finding the number of unique categories and checking null/dummy values for each of the working data.

In [19]:
for col_name in df_work.columns:
    if df_work[col_name].dtypes == 'object':
        unique_cat = len(df_work[col_name].unique())
        print("'{col_name}' has {unique_cat} unique categories".format(col_name=col_name, unique_cat=unique_cat))

'productID' has 4181 unique categories
'reviewText' has 99928 unique categories
'reviewerID' has 60813 unique categories


In [17]:
# Checking for dummy values in the data set

df_work.isnull().sum().sort_values(ascending=False).head()

reviewerID    0
reviewText    0
overall       0
productID     0
dtype: int64

Writing this preprocessed dataframe into a csv file to work from.

In [18]:
df_work.to_csv('Electronics_work.csv')

In [20]:
df_work.head()

productID  overall                                         reviewText  \
1  0528881469        1  I'm a professional OTR truck driver, and I bou...   
2  0528881469        3  Well, what can I say.  I've had this unit in m...   
3  0528881469        2  Not going to write a long review, even thought...   
4  0528881469        1  I've had mine for a year and here's what we go...   
5  0594451647        5  I am using this with a Nook HD+. It works as d...   

       reviewerID  
1   AMO214LNFCEI4  
2  A3N7T0DY83Y4IG  
3  A1H8PY3QHMQQA0  
4  A24EV6RXELQZ63  
5  A2JXAZZI9PHK9Z